node/impl/full/eth.go

```diff
@@ -1270,7 +1271,33 @@ func (a *EthModule) EthCall(ctx context.Context, tx ethtypes.EthCall, blkParam e
 	return ethtypes.EthBytes{}, nil
 }
 
+func (e *EthEventHandler) GetEthLogsForBlockAndTransaction(ctx context.Context, blockHash *ethtypes.EthHash, txHash ethtypes.EthHash) ([]ethtypes.EthLog, error) {
+	ces, err := e.ethGetEventsForFilter(ctx, &ethtypes.EthFilterSpec{BlockHash: blockHash})
+	if err != nil {
+		return nil, err
+	}
+	logs, err := ethFilterLogsFromEvents(ctx, ces, e.SubManager.StateAPI)
+	if err != nil {
+		return nil, err
+	}
+	var out []ethtypes.EthLog
+	for _, log := range logs {
+		if log.TransactionHash == txHash {
+			out = append(out, log)
```

Comment:
@rvagg Doing it this way so we don't need to add one more "msgCid" index to the events Index given that we just optimised perf based on using the height and tip set cid indices.

This will utilise the `tipset_cid` index we already have on the Event Index.

Reply:
Fair enough.

Also if we're using the filter API then we don't have any other choices - however we could take this further if we wanted to - `EventFilterManager#Install` could take a message CID argument and we could build that into the query without adding an index for it, it'll still hit the tipset index but it'll do this same filtering in the sqlite layer rather than having to bubble all of the data up through the cgo interface, through the event object creation and through the eth translation layer too just to get filtered out here.

Maybe not a bit enough deal to bother with now? But it could be a big difference on tipsets with a _lot_ of events and a lot of messages. Maybe add a TODO here to note that this filtering could be done in the sql layer for "msgCid".

Reply:
Done.